### PR TITLE
{Packaging} Bump azure-core from 1.21.1 to 1.24.0

### PR DESCRIPTION
**Pull request by ZengTaoxu**
**Related command**

**Description**

Some modules (https://github.com/Azure/azure-cli/pull/22486) and extensions (AML) which use new SDK need newer version of azure-core, so we need upgrade it to the latest version (1.24.0).

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Review by jsntcy — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/synapse/tests/latest/test_synapse_scenario.py [hunk 1]
```diff
@@ -1406,12 +1406,13 @@ def test_sql_pool(self):
         self.cmd('az synapse sql pool show --name {sql-pool} --workspace {workspace} --resource-group {rg}',
                  expect_failure=True)
 
-    @record_only()
+    @unittest.skip('Resource group \'rgtesting\' could not be found')
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/synapse/tests/latest/test_synapse_scenario.py:1409]: Skip it temporarily as we cannot pass test lively as the test used fixed resources, will contact service team to fix it later.

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py [hunk 2]
```diff
@@ -6744,7 +6744,7 @@ def test_aks_create_edge_zone(self, resource_group, resource_group_location):
         self.test_resources_count = 0
         # kwargs for string formatting
         aks_name = self.create_random_name('cliakstest', 16)
-        kubernetes_version = "1.20.7"
+        kubernetes_version = "1.23.5"
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py:6747]: Just of curiosity, may I ask why do we need to upgrade this `kubernetes_version` ?

**Review by zhoxing-ms — COMMENTED** [review]
Could you add the PR description to explain the background and context (such as the related issue/PR link) of bumping `azure-core` version? 
This is convenient for us to track the cause and impact of this PR in the future

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py (hunk 2 above)
Comment (ZengTaoxu) [reply src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py:6747]: Because version 1.20.7 is not supported in this region.

**Review by evelyn-ys — APPROVED** [review]

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**